File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {

        vector<vector<int>> res;
        set<vector<int>> st;

        
        for(int i = 0; i< nums.size(); i++) {
            for(int j =  i + 1; j < nums.size(); j++) {
                for(int k = j + 1; k < nums.size(); k++) {
                    if(i != j && i != k && j != k) {
                        if(nums[i] + nums[j] + nums[k] == 0) {
                            vector<int> triplet = {nums[i], nums[j], nums[k]};
                            vector<int> sorted_triplet = triplet;
                            sort(sorted_triplet.begin(), sorted_triplet.end());
                            if(!(st.contains(sorted_triplet))) {
                                res.push_back(triplet);
                                sort(triplet.begin(), triplet.end());
                                st.insert(triplet);
                            }
                            
                        }
                    }
                }
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/concatenation-of-array/submission-2.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        #TC: O(2n), SC: O(2n)
        ans = [0] * (2 * len(nums)) 
        # for i in range(2):
        for j in range(len(nums)):
            ans[j] = nums[j]
            ans[j + len(nums)] = nums[j]
        
        return ans